### ISSUE TITLE
save option not working

### ISSUE THREAD

**Issue by fireosul** [mypaint/mypaint#370]
Any one having trouble on my paint where they can't save. Its says error. I keep trying to save as different files but it is not working.

**Comment by achadwick** [comment]
(Thanks for the feedback! Please read the [bug reporting guidelines](https://github.com/mypaint/mypaint/blob/master/CONTRIBUTING.md#test-the-program-and-report-issues) so we can deal with this issue more effectively.)

* I need to know what version of MyPaint this is, and from where you got it. You can find the version in Help→About MyPaint.
* I need to know what OS (and version) you are running MyPaint on.
* I need to know the full error text you are referring to. Please paste it here.

**Comment by fireosul** [comment]
The program I have is  mypaint-1.0.0-win32-installer.exe* ( build by tumagonx ) I got this from the my paint website I'm running this on Microsoft windows XP The save option wont work. I keep getting an error message that tells me "Unable to save: permission denied." and another pop up comes up saying there is a programming bug detected. 


     On Tuesday, July 21, 2015 6:17 PM, Andrew Chadwick <notifications@github.com> wrote:
   

 (Thanks for the feedback! Please read the bug reporting guidelines so we can deal with this issue more effectively.)   
   - I need to know what version of MyPaint this is, and from where you got it. You can find the version in Help→About MyPaint.
   - I need to know what OS (and version) you are running MyPaint on.
   - I need to know the full error text you are referring to. Please paste it here.

**Comment by achadwick** [comment]
Sorry, but 1.0.0 is super-old, and we don't support it any more.

Also, support for Windows XP has been dropped by Microsoft [since April 2014](https://en.wikipedia.org/wiki/Windows_XP#End_of_support). You should upgrade your operating system to Windows 7 or newer: I don't intend to support MyPaint under Windows Vista or earlier.

A newer version of MyPaint is available. Please could you try either the new 1.2.0-beta build from <https://github.com/mypaint/mypaint/releases/tag/v1.2.0-beta.0>, or @tumagonx 's rather better tested 1.2.0-alpha build from <http://www.opensourcepack.blogspot.fr/2013/01/mypaint-and-pygi.html>? Let us know if this has not fixed the problem. Either *might* work under obsolete versions of Windows, but this is untested.

**Comment by achadwick** [comment]
**Triage note:** Obsolete versions. Needs backtrace. Needs steps to reproduce. May need info about the reporter's system (filesystem encoding issues?). If this is reproducible with Win 7 or later and a recent alpha or beta release for 1.2.0, this may indicate a fault we can fix. However, close in 2 weeks if no actionable report emerges.

**Comment by zb13y** [comment]
Save and load. Just error message, no problems with it, idk what it means. MyPaint 1.2.0 Windows 7 64.
   
OPEN
   ```python
Mypaint version: 1.2.0-beta.0+gitexport.adc9031
System information: Windows-7-6.1.7601-SP1
Traceback (most recent call last):
  File "C:\Program Files\MyPaint\share\mypaint/gui/filehandling.py", line 492, update_preview_cb(self=<gui.filehandling.FileHandler object>, file_chooser=<FileChooserDialog object at 0xd157af0 (GtkFileChooserDialog at 0xaaea580)>, 

preview=<Image object at 0xd157c30 (GtkImage at 0xd7a7900)>)
                filename = filename.decode('utf-8')
                pixbuf = helpers.freedesktop_thumbnail(filename)
                if pixbuf:
  variables: {'helpers.freedesktop_thumbnail': ('global', <function freedesktop_thumbnail at 0x000000000a128488>), 'pixbuf': (None, []), 'filename': ('local', u'\\Games')}
  File "C:\Program Files\MyPaint\share\mypaint/lib/helpers.py", line 167, freedesktop_thumbnail(filename=u'\\Games', pixbuf=None)
        uri = _filename2uri_freedesktop_canon(os.path.abspath(filename))
        logger.debug("thumb: uri=%r", uri)
  variables: {'os.path.abspath': ('global', <function abspath at 0x00000000024c29b0>), '_filename2uri_freedesktop_canon': ('global', <function _filename2uri_freedesktop_canon at 0x000000000a128668>), 'uri': (None, []), 'filename': 

('local', u'\\Games')}
  File "C:\Program Files\MyPaint\share\mypaint/lib/helpers.py", line 308, _filename2uri_freedesktop_canon(path=u'C:\\Games', encoding='mbcs')
        path_bytes = path.encode(encoding)
        return GLib.filename_to_uri(path_bytes, None)
  variables: {'None': ('builtin', None), 'path_bytes': ('local', 'C:\\Games'), 'GLib.filename_to_uri': ('global', gi.FunctionInfo(filename_to_uri))}
Error: g-invoke-error-quark: Could not locate g_filename_to_uri: 'g_filename_to_uri': Не найдена указанная процедура. (1)

SAVE
    ```python
Mypaint version: 1.2.0-beta.0+gitexport.adc9031
System information: Windows-7-6.1.7601-SP1
Traceback (most recent call last):
  File "C:\Program Files\MyPaint\share\mypaint/gui/filehandling.py", line 608, save_as_cb(self=<gui.filehandling.FileHandler object>, action=<Action object at 0xae46a00 (GtkAction at 0xadfc8e0)>)
            else:
                self.save_as_dialog(self.save_file, suggested_filename=current_filename)
  variables: {'suggested_filename': (None, []), 'self.save_file': ('local', <bound method FileHandler.save_file of <gui.filehandling.FileHandler object at 0x000000000ae47410>>), 'self.save_as_dialog': ('local', <bound method 

FileHandler.save_as_dialog of <gui.filehandling.FileHandler object at 0x000000000ae47410>>), 'current_filename': ('local', '')}
  File "C:\Program Files\MyPaint\share\mypaint/gui/filehandling.py", line 670, save_as_dialog(self=<gui.filehandling.FileHandler object>, save_method_reference=<bound method FileHandler.save_file of <gui.filehandling.FileHandler 

object>>, suggested_filename='', start_in_folder=None, export=False, **options={})
                        else:
                            save_method_reference(filename, **options)
                        break
  variables: {'options': ('local', {}), 'save_method_reference': ('local', <bound method FileHandler.save_file of <gui.filehandling.FileHandler object at 0x000000000ae47410>>), 'filename': ('local', u'\\Games\\111.ora')}
  File "C:\Program Files\MyPaint\share\mypaint/gui/drawwindow.py", line 83, wrapper(self=<gui.filehandling.FileHandler object>, *args=(u'\\Games\\111.ora',), **kwargs={})
            try:
                func(self, *args, **kwargs)
                # gtk main loop may be called in here...
  variables: {'self': ('local', <gui.filehandling.FileHandler object at 0x000000000ae47410>), 'args': ('local', (u'\\Games\\111.ora',)), 'func': ('local', <function save_file at 0x000000000ad80578>), 'kwargs': ('local', {})}
  File "C:\Program Files\MyPaint\share\mypaint/gui/filehandling.py", line 380, save_file(self=<gui.filehandling.FileHandler object>, filename=u'\\Games\\111.ora', export=False, **options={})
                thumbnail_pixbuf = self.doc.model.render_thumbnail(**options)
            helpers.freedesktop_thumbnail(filename, thumbnail_pixbuf)
  variables: {'thumbnail_pixbuf': ('local', <Pixbuf object at 0xfc415a0 (GdkPixbuf at 0xd0e1600)>), 'helpers.freedesktop_thumbnail': ('global', <function freedesktop_thumbnail at 0x000000000a1e2488>), 'filename': ('local', u'\\Games\

\111.ora')}
  File "C:\Program Files\MyPaint\share\mypaint/lib/helpers.py", line 167, freedesktop_thumbnail(filename=u'\\Games\\111.ora', pixbuf=<Pixbuf object at 0xfc415a0 (GdkPixbuf at 0xd0e1600)>)
        uri = _filename2uri_freedesktop_canon(os.path.abspath(filename))
        logger.debug("thumb: uri=%r", uri)
  variables: {'os.path.abspath': ('global', <function abspath at 0x00000000024c29b0>), '_filename2uri_freedesktop_canon': ('global', <function _filename2uri_freedesktop_canon at 0x000000000a1e2668>), 'uri': (None, []), 'filename': 

('local', u'\\Games\\111.ora')}
  File "C:\Program Files\MyPaint\share\mypaint/lib/helpers.py", line 308, _filename2uri_freedesktop_canon(path=u'C:\\Games\\111.ora', encoding='mbcs')
        path_bytes = path.encode(encoding)
        return GLib.filename_to_uri(path_bytes, None)
  variables: {'None': ('builtin', None), 'path_bytes': ('local', 'C:\\Games\\111.ora'), 'GLib.filename_to_uri': ('global', gi.FunctionInfo(filename_to_uri))}
Error: g-invoke-error-quark: Could not locate g_filename_to_uri: 'g_filename_to_uri': Не найдена указанная процедура. (1)

**Comment by achadwick** [comment]
@zb13y I don't know if that is the same issue that @fireosul saw. Please could you raise it as a separate issue? Thanks.

(I think some of the Traceback may have been incorrectly pasted - it seems a little garbled somehow?)

**Comment by achadwick** [comment]
@fireosul Please test with the updated beta (v1.2.0-beta.0.73) just released at <https://github.com/mypaint/mypaint/releases/tag/v1.2.0-beta.0>. Does that fix this issue for you? Please respond.

**Comment by achadwick** [comment]
Many bugs like this during save have been fixed recently, including ones that present as permissions errors. MyPaint was mishandling a variable containing a path, doing character conversions wrongly. Attempting to write into a garbled path that doesn't exist in a protected location, such as `C:\Users\<garble>\your\chosen\path.ora`, presents as a permissions error on Windows.

@fireosul 
My assumption is that you had a non-ASCII character in the path you were saving to, and that this was the cause of your error in the earlier release. I don't have much else to go on still, sorry.

You reported this, and are affected by it. I believe it has been fixed in the most recent beta release. Are you able to test that? I'd love to get some positive "it's fixed" feedback!

New versions here: <https://github.com/mypaint/mypaint/releases/>

**Comment by achadwick** [comment]
**Triage note**: believed fixed, close in 7 days if no feedback.

**Comment by achadwick** [comment]
I believe this problem has been fixed in the recent beta releases (1.2.0-beta.0 and beyond). Closing the issue now: people looking for this issue will still be able to search for it.

Anybody still experiencing this in 1.2.0-beta.1 or later should either raise a new issue report referring to this one, or reopen this one with good feedback. Thanks.
